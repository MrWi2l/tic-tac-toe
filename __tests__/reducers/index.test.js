import {
  changeSquares,
  changePlayer,
} from '../../src/reducers/index'

describe('CHANGE SQUARE REDUCER', () => {
  it('should return the initial state', () => {
    expect(
      changeSquares(undefined, {}),
    ).toMatchSnapshot()
  })

  it('should return the new state with squares initialize', () => {
    expect(
      changeSquares([], {
        type: 'INITIALIZE_SQUARES',
      }),
    ).toMatchSnapshot()
  })

  it('should return the new state with square updated', () => {
    expect(
      changeSquares(Array(9).fill(null), {
        type: 'ADD_SQUARE',
        index: 1,
        symbol: 'X',
      }),
    ).toMatchSnapshot()
  })
})

describe('CHANGE PLAYER REDUCER', () => {
  it('should return the initial state', () => {
    expect(
      changePlayer(undefined, {}),
    ).toMatchSnapshot()
  })

  it('should return the initial player', () => {
    expect(
      changePlayer(undefined, { type: 'INITIALIZE_PLAYER' }),
    ).toMatchSnapshot()
  })

  it('should return the opposite of the current player (X)', () => {
    expect(
      changePlayer(undefined, { type: 'CHANGE_PLAYER', currentPlayer: 'X' }),
    ).toMatchSnapshot()
  })
})
