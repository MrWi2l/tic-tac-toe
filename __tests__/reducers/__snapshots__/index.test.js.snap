exports[`CHANGE PLAYER REDUCER should return the initial player 1`] = `"X"`;

exports[`CHANGE PLAYER REDUCER should return the initial state 1`] = `""`;

exports[`CHANGE PLAYER REDUCER should return the opposite of the current player (X) 1`] = `"O"`;

exports[`CHANGE SQUARE REDUCER should return the initial state 1`] = `Array []`;

exports[`CHANGE SQUARE REDUCER should return the new state with square updated 1`] = `
Array [
  null,
  "X",
  null,
  null,
  null,
  null,
  null,
  null,
  null,
]
`;

exports[`CHANGE SQUARE REDUCER should return the new state with squares initialize 1`] = `
Array [
  null,
  null,
  null,
  null,
  null,
  null,
  null,
  null,
  null,
]
`;
